segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/AsyncBaseChunkStorage.java

```diff
@@ -142,16 +142,16 @@ public AsyncBaseChunkStorage(Executor executor) {
 
         // Call concrete implementation.
         val returnFuture = doCreateAsync(chunkName);
-        val metricsFuture = returnFuture.thenAcceptAsync(handle -> {
+        returnFuture.thenAcceptAsync(handle -> {
             // Record metrics.
             val elapsed = timer.getElapsed();
             ChunkStorageMetrics.CREATE_LATENCY.reportSuccessEvent(elapsed);
             ChunkStorageMetrics.CREATE_COUNT.inc();
             log.debug("Create - chunk={}, latency={}.", chunkName, elapsed.toMillis());
+            if (log.isTraceEnabled()) {
```

Comment:
this check is not needed. `log.trace` will be a no-op in that case. The whole point of doing the check originally was not to add yet another callback to the Future. In this refactored code, it is no longer an issue.

Reply:
same below